### PR TITLE
[12.0] ADD product_model_viewer

### PR DESCRIPTION
**Pull request by eLBati**
See https://github.com/OCA/web/tree/12.0/web_widget_model_viewer

### CODE REVIEW

**Comment by pedrobaeza** [comment]
Can you provide demo data for showcasing both modules?

**Comment by eLBati** [comment]
@pedrobaeza right, done

**Review by mamcode — APPROVED** [review]
LGTM, very good contribution.

**Review by marcelsavegnago — APPROVED** [review]

**Comment by pedrobaeza** [comment]
Several thoughts:

- You are linking the 3D model to the template, but there can be the option of having a different 3D model for each variant. I think you should handle this the same way as product images, defaulting to the template image, but having the option to put a different one in the variant.
- Does this worth to be split from `website_sale_product_model_viewer`? If you plan to do another modules for any feature in backend or without `website_sale`, then it can be, but if the only idea is to put it on the e-commerce, then not sure. If not split, I would move the field to the e-commerce tab.
- It's worth to mention in the README in the usage section that with demo data, the product "Office Chair Black" contains a sample 3D model.

**Comment by mamcode** [comment]
Agree with @pedrobaeza  in all but I think that split with `website_sale_product_model_viewer` is good because there may be projects without e-commerce that only need to have the 3D model in the product in the backend.

**Comment by andreampiovesana** [comment]
I think that we have to emulate the image, so is better to link the 3d to the product.product model?

**Comment by eLBati** [comment]
@pedrobaeza @mamcode @andreampiovesana implemented variant handling, similarly to images.
Added `USAGE.rst`.

I kept `product_model_viewer`splitted from `website_sale_product_model_viewer` for the reasons explained by @mamcode . The 3D model of products could be also used by other modules, not related to e-commerce.

**Comment by pedrobaeza** [comment]
OK with the split for future uses.

File: product_model_viewer/views/product_views.xml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8"?>
+<odoo>
+    <record id="product_template_form_view_3d" model="ir.ui.view">
+        <field name="name">product_template_form_view_3d</field>
+        <field name="model">product.template</field>
+        <field name="inherit_id" ref="product.product_template_form_view"/>
+        <field name="arch" type="xml">
+            <xpath expr="//notebook" position="inside">
+                <page string="3D model">
```
Comment (bealdav) [inline product_model_viewer/views/product_views.xml]: Congratulations for this really nice module

Completely expressive like that, I think (only my opinion)
```suggestion
                <page string="3D">
```
When you have many tabs, shorter ones are good ;-)

Comment (eLBati) [reply product_model_viewer/views/product_views.xml]: Thanks, done

**Review by andreampiovesana — APPROVED** [review]

**Review by bealdav — APPROVED** [review]

**Comment by rousseldenis** [comment]
@eLBati Ready ?

**Comment by eLBati** [comment]
@rousseldenis this is ready for me, thanks

**Pull request merged**